targettests/execution/from_state_complex.cpp

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ %cpp_std --enable-mlir %s -o %t && %t | FileCheck %s
```

Comment:
How does this test that the state decomposition pass has run? Correct me if I'm wrong, but as-is I think this just compiles with MLIR and runs against the simulator, and the RemoteRESTQPU is never used. Do you not have to test with something like `--target quantinuum --emulate` ? 

I think it would also be good to have a FileCheck test in the `tests/Quake` folder that explictly tests that the init_state op was decomposed into quake quantum operations.

Reply:
Will do, thanks!

Reply:
Updated the test and also added a test in tests/Quake